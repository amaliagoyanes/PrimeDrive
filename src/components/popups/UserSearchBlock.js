import React, {Component} from 'react';
import {View,} from 'react-native';
import {Row, Col} from 'react-native-easy-grid';
import SwipeALot from 'react-native-swipe-a-lot';
import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome';
import {Hideo} from 'react-native-textinput-effects';
import * as Progress from 'react-native-progress';

import {IconButton,} from '../index';
import styles from '../../style/styles';
import plainStyles from '../../style/plain';
import {scale, scaleText} from '../../scaling';


export default class UserSearchBlock extends Component {
    constructor(props) {
        super(props);
        this.state = {
            data: [], search: ''
        };

        this.getItems = this.getItems.bind(this);
    }

    getItems() {
        let users = Object.keys(this.props.data).map(key => {
            return {...this.props.data[key], uid: key}
        });
        if (this.state.search !== '')
            users = users.filter((obj) => {
                return obj.name.toLowerCase().indexOf(this.state.search.toLowerCase()) !== -1 || obj.number.toString().indexOf(this.state.search) !== -1
            });


        let perRow = 4,
            rows = 4,
            perSlide = 16,
            slides = Math.ceil(users.length / perSlide);
        return Array.apply(null, {length: slides}).map((d, idxSld) =>
            <Col style={[styles.slide, {marginBottom: 0}]} key={idxSld}>
                {Array.apply(null, {length: rows}).map((d, idxRow) =>
                    <Row key={idxRow}>
                        {Array.apply(null, {length: perRow}).map((d, idxItem) => {
                            let item = users[idxSld * (perSlide - 1) + idxRow * rows + idxItem];
                            if (item) {
                                return <IconButton key={idxItem} text={item.name}
                                                   price={item.number.toString()} active
                                                   onPress={() => this.props.login(item.number)}/>
                            } else {
                                return <Col key={idxItem}/>
                            }
                        })}
                    </Row>
                )}
                <Row size={0.1}/>
            </Col>
        )
    }

    render() {
        return (
            <Col size={6} style={styles.userLoginLeft}>
                <Row size={1} style={{paddingRight: scale(5), paddingLeft: scale(5)}}>
                    <Col>
                        <Hideo
                            style={{alignItems: 'center', justifyContent: 'center'}}
                            label={'Search'}
                            iconClass={FontAwesomeIcon}
                            iconName={'search'}
                            iconColor={'black'}
                            iconBackgroundColor={'transparent'}
                            inputStyle={{
                                color: 'black',
                                backgroundColor: 'transparent',
                                fontSize: scaleText(20)
                            }}
                            onChangeText={text => this.setState({search: text}) }
                        />
                    </Col>
                </Row>
                <Row size={8}>
                    {(!Object.keys(this.props.data).length) ?
                        <View style={{flexDirection: 'row', justifyContent: 'center', flex: 1, alignItems: 'center'}}>
                            <Progress.Circle borderWidth={5} borderColor={"#2dab61"} size={scale(60)} indeterminate={true} />
                        </View> :
                        <SwipeALot circleDefaultStyle={{...plainStyles.dotStyleModal, top: scale(5)}}
                                   circleActiveStyle={plainStyles.activeDotStyleModal}>
                            {this.getItems()}
                        </SwipeALot>
                    }
                </Row>
            </Col>
        )
    }
}
