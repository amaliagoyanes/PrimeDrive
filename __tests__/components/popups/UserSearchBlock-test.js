'use strict';

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import UserSearchBlock from '../../../src/components/popups/UserSearchBlock';

UserSearchBlock.prototype.componentDidMount = jest.fn;

let state = {data: [{name: "Jens", number: 1}, {name: "Test", number: 2}]};

it('renders correctly', () => {
    const tree = shallow(
        <UserSearchBlock />);
    tree.setState(state);
    expect(tree).toMatchSnapshot();
    expect(tree.find('Hideo').length).toEqual(1);
    expect(tree.find('SwipeALot').length).toEqual(1);
});

it('filters correctly name', () => {
    const tree = shallow(
        <UserSearchBlock />);
    tree.setState(state);
    tree.setState({search: "Jens"});
    expect(tree).toMatchSnapshot();
});

it('filters correctly number', () => {
    const tree = shallow(
        <UserSearchBlock />);
    tree.setState(state);
    tree.setState({search: '1'});
    expect(tree).toMatchSnapshot();
});

