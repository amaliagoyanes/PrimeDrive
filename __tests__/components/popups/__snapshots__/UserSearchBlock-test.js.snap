// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filters correctly name 1`] = `
<ColumnNB
  size={6}
  style={
    Object {
      "borderColor": "#404048",
      "borderRightWidth": 0.2,
      "marginBottom": 20,
      "paddingRight": 30,
    }
  }
>
  <RowNB
    size={1}
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <ColumnNB>
      <Hideo
        animationDuration={200}
        height={48}
        iconBackgroundColor="transparent"
        iconClass={[Function]}
        iconColor="black"
        iconName="search"
        inputStyle={
          Object {
            "backgroundColor": "transparent",
            "color": "black",
            "fontSize": 20,
          }
        }
        label="Search"
        onChangeText={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
    </ColumnNB>
  </RowNB>
  <RowNB
    size={8}
  >
    <SwipeALot
      circleActiveStyle={
        Object {
          "backgroundColor": "#404048",
        }
      }
      circleDefaultStyle={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#404048",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 8,
          "margin": 7,
          "marginTop": 3,
          "top": 5,
          "width": 8,
        }
      }
    >
      <ColumnNB
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
            },
            Object {
              "marginBottom": 0,
            },
          ]
        }
      >
        <RowNB>
          <IconButton
            active={true}
            onPress={[Function]}
            price="1"
            text="Jens"
          />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB
          size={0.1}
        />
      </ColumnNB>
    </SwipeALot>
  </RowNB>
</ColumnNB>
`;

exports[`filters correctly number 1`] = `
<ColumnNB
  size={6}
  style={
    Object {
      "borderColor": "#404048",
      "borderRightWidth": 0.2,
      "marginBottom": 20,
      "paddingRight": 30,
    }
  }
>
  <RowNB
    size={1}
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <ColumnNB>
      <Hideo
        animationDuration={200}
        height={48}
        iconBackgroundColor="transparent"
        iconClass={[Function]}
        iconColor="black"
        iconName="search"
        inputStyle={
          Object {
            "backgroundColor": "transparent",
            "color": "black",
            "fontSize": 20,
          }
        }
        label="Search"
        onChangeText={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
    </ColumnNB>
  </RowNB>
  <RowNB
    size={8}
  >
    <SwipeALot
      circleActiveStyle={
        Object {
          "backgroundColor": "#404048",
        }
      }
      circleDefaultStyle={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#404048",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 8,
          "margin": 7,
          "marginTop": 3,
          "top": 5,
          "width": 8,
        }
      }
    >
      <ColumnNB
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
            },
            Object {
              "marginBottom": 0,
            },
          ]
        }
      >
        <RowNB>
          <IconButton
            active={true}
            onPress={[Function]}
            price="1"
            text="Jens"
          />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB
          size={0.1}
        />
      </ColumnNB>
    </SwipeALot>
  </RowNB>
</ColumnNB>
`;

exports[`renders correctly 1`] = `
<ColumnNB
  size={6}
  style={
    Object {
      "borderColor": "#404048",
      "borderRightWidth": 0.2,
      "marginBottom": 20,
      "paddingRight": 30,
    }
  }
>
  <RowNB
    size={1}
    style={
      Object {
        "paddingLeft": 5,
        "paddingRight": 5,
      }
    }
  >
    <ColumnNB>
      <Hideo
        animationDuration={200}
        height={48}
        iconBackgroundColor="transparent"
        iconClass={[Function]}
        iconColor="black"
        iconName="search"
        inputStyle={
          Object {
            "backgroundColor": "transparent",
            "color": "black",
            "fontSize": 20,
          }
        }
        label="Search"
        onChangeText={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      />
    </ColumnNB>
  </RowNB>
  <RowNB
    size={8}
  >
    <SwipeALot
      circleActiveStyle={
        Object {
          "backgroundColor": "#404048",
        }
      }
      circleDefaultStyle={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "#404048",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 8,
          "margin": 7,
          "marginTop": 3,
          "top": 5,
          "width": 8,
        }
      }
    >
      <ColumnNB
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "marginBottom": 15,
            },
            Object {
              "marginBottom": 0,
            },
          ]
        }
      >
        <RowNB>
          <IconButton
            active={true}
            onPress={[Function]}
            price="1"
            text="Jens"
          />
          <IconButton
            active={true}
            onPress={[Function]}
            price="2"
            text="Test"
          />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB>
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
          <ColumnNB />
        </RowNB>
        <RowNB
          size={0.1}
        />
      </ColumnNB>
    </SwipeALot>
  </RowNB>
</ColumnNB>
`;
